keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -443,107 +453,173 @@ General prep:
 
 Here is the workflow:
 
-+ A `Bucket` instance created by COSI must be cloned by the admin (each `Bucket` instance needs a unique name).
-+ The user creates a `BucketRequest` (BR) that specifies the `Bucket` instance name (BR.bucketInstanceName) provided by the admin.
-+ The user creates a `BucketAccessRequest` (BAR) - see [grant bucket access](#grant-bucket-access) -- and the BAR references their BR.
++ The user creates a `BucketAccessRequest` (BAR) that specifies either the name of the BR (when sharing in the same namespace) or the `Bucket` instance name (when sharing across namespaces).
++ COSI sees the BAR add event.
+   + Note: the `Bucket` instance name is referenced either as BAR.BR.bucketName or BAR.bucketName.
++ COSI checks that the `Bucket` instance has a nil deletionTimestamp.
++ COSI adds the BAR finalizer.
++ COSI creates a BA with a one-way binding to the `Bucket` instance.
++ COSI adds a BA finalizer.
++ The sidecar sees the new BA and calls the driver to grant access to the backend bucket.
+   + Note: the backend bucket is found via BA.bucketName.bucketID.
++ COSI updates the BA and BAR statuses to "accessGranted".
++ COSI adds a `Bucket` finalizer, "objectstorage.k8s.io/ba-\<name>".
+
+In addition, a pod references a BAR in the `volumes.csi.driver` spec:
++ The cosi-node-adapter receives a _NodeStageVolume_ request from the kubelet.
++ The adapter adds a "objectstorage.k8s.io/pod-\<name>" finalizer to the associated `BucketAccess`.
 
 ## Delete Bucket
 
-![DeleteBucket Workflow](images/delete-bucket.png)
+![DeleteBucket Workflow](images/delete-bucket.pni)
 
-This workflow describes the automation designed for deleting a `Bucket` instance and optionally the related backend bucket. Revoking access to this bucket is covered in [Revoke Bucket Access](#revoke-bucket-access).
+This workflow describes optionally deleting a `Bucket` instance and the backend bucket. Revoking access to this bucket is covered in [Revoke Bucket Access](#revoke-bucket-access). There are three deletion policies: "Retain", "Delete", and "ForceDelete", which are defined in the `BucketClass` and the `Bucket` instance. Only the bucket instance's _deletionPolicy_ is used to determine the deletion policy used to handle bucket deletes.
 
-It is important to understand that when a `BucketRequest` deletion triggers the delete of the associated `Bucket` instance, the `Bucket` instance is deleted regardless of access. This means that even if a `BucketAccess` references a `Bucket`, the `Bucket is still deleted and the BA is orphaned. Additionally, even if multiple `Bucket` instances point to the same backend bucket, if the retention policy is "Delete" then the bucket is deleted and workloads may fail. Potentially, this may only apply to MVP.
+It is up to each provisioner whether or not to physically delete the backend bucket content. But, for idempotency, the expectation is that the backend bucket will at least be made unavailable.
```

Comment:
Nit: bucket content isn't really a physical thing even though we talk about it metaphorically like it is. I could imagine this being confusing for non-native speakers.

```suggestion
It is up to each provisioner whether or not to delete the backend bucket content. But, for idempotency, the expectation is that the backend bucket will at least be made unavailable.
```